data_structures/linked_list/circular_doubly_linked_list.c

```diff
@@ -0,0 +1,307 @@
+    (temp1->next)->prev = temp2;
+    head = temp1->next;
+    free(temp1);
+    temp1 = NULL;
+    printf("\nNode succesfully deleted\n\n");
+    return head;
+}
+
+/**
+ * @brief Function for deletion of the last node in list
+ *
+ * @param head      starting pointer of list
+ * @return ListNode*
+ */
+ListNode *delete_last_node(ListNode *head)
+{
+    if (head == NULL)
+    {
+        printf("list is Empty\n");
+        return head;
+    }
+
+    ListNode *temp1, *temp2;
+    temp1 = head;
+    temp2 = temp1->prev;
+    if (temp1 == temp2)
+    {
+        free(temp2);
+        head = NULL;
+        printf("\nNode succesfully deleted\n\n");
+        return head;
+    }
+    (temp2->prev)->next = temp1;
+    temp1->prev = temp2->prev;
+    free(temp2);
+    temp2 = NULL;
+    printf("\nNode succesfully deleted\n\n");
+    return head;
+}
+
+/**
+ * @brief The function to reverse the linked list
+ *
+ * @param head      starting pointer of list
+ * @return ListNode*
+ */
+ListNode *reverseLL(ListNode *head)
+{
+    if (!head)
+        return NULL;
+
+    ListNode *new_head = NULL;
+    ListNode *last = head->prev;
+    ListNode *curr = last, *prev;
+
+    // traverse list in backward direction
+    while (curr->prev != last)
+    {
+        prev = curr->prev;
+        new_head = insert_at_last(new_head, curr->value);
+        curr = prev;
+    }
+    new_head = insert_at_last(new_head, curr->value);
+
+    return new_head;
+}
+
+/**
+ * @brief The function that will return current size of list
+ *
+ * @param head      starting pointer of list
+ * @return int      size of list
+ */
+int getsize(ListNode *head)
+{
+    int size = 1;
+    ListNode *temp = head->next;
+    while (temp != head)
+    {
+        temp = temp->next;
+        size++;
+    }
+    return size;
+}
+
+/**
+ * @brief   Display list function
+ * @param   head   starting pointer of list
+ * @returns void
+ */
+
+void display(ListNode *head)
+{
+    printf("\nIts Your Linked-list\n\n");
+    ListNode *temp;
+    temp = head;
+    if (head != NULL)
+    {
+        while (temp->next != head)
+        {
+            printf("%lld -> ", temp->value);
+            temp = temp->next;
+        }
+        if (temp->next == head)
+        {
+            printf("%lld", temp->value);
+        }
+    }
+    else
+    {
+        printf("list is Empty\n");
+    }
+    printf("\n");
+}
+
+/**
+ * @brief Main function
+ * @returns 0 on exit
+ */
+int main()
+{
+    // Creating an empty list
+    ListNode *head = NULL;
+    // Flg to stop loop and value to insert
+    int flag = 1;
+    uint64_t value = 0;
+    while (flag == 1)
+    {
+        int ch;
+        printf("\nWhat do you want to perform?\n");
+        printf(
+            "ENTER YOUR CHOICE\n1. Insert At head\n2. Insert At End\n3. "
+            "Delete First node \n4. Delete Last Node \n5. Reverse List\n6. Get "
+            "List size\n7. Display List"
+            "\n8. Exit\n");
+        scanf("%d", &ch);
```

Comment:
Since your code accepts user input and you said you are still learning, I should put some interesting reading materials here, [about scanf](https://sekrit.de/webdocs/c/beginners-guide-away-from-scanf.html), [alternatives to scanf](https://stackoverflow.com/questions/58403537/what-can-i-use-for-input-conversion-instead-of-scanf)